The ARM build errors are:

```
/__w/1/s/src/coreclr/src/pal/src/misc/sysinfo.cpp:578:12: error: no matching function for call to 'ReadMemoryValueFromFile'
        if(ReadMemoryValueFromFile("/sys/devices/system/cpu/cpu0/cache/index0/size", &size))
           ^~~~~~~~~~~~~~~~~~~~~~~
/__w/1/s/src/coreclr/src/pal/src/misc/sysinfo.cpp:514:1: note: candidate function not viable: no known conversion from 'size_t *' (aka 'unsigned int *') to 'uint64_t *' (aka 'unsigned long long *') for 2nd argument
ReadMemoryValueFromFile(const char* filename, uint64_t* val)
...
```

--- a/src/coreclr/src/gc/unix/gcenv.unix.cpp
+++ b/src/coreclr/src/gc/unix/gcenv.unix.cpp
@@ -857,7 +857,7 @@ static size_t GetLogicalProcessorCacheSizeFromOS()
     }
 #endif
 
-#if defined(HOST_ARM64)
+#if defined(TARGET_LINUX) && !defined(HOST_ARM)
     if (cacheSize == 0)
     {
         // It is currently expected to be missing cache size info

--- a/src/coreclr/src/gc/unix/gcenv.unix.cpp
+++ b/src/coreclr/src/gc/unix/gcenv.unix.cpp
@@ -834,7 +834,7 @@ static size_t GetLogicalProcessorCacheSizeFromOS()
     cacheSize = std::max(cacheSize, ( size_t) sysconf(_SC_LEVEL4_CACHE_SIZE));
 #endif
 
-#if defined(TARGET_LINUX)
+#if defined(TARGET_LINUX) && !defined(HOST_ARM)
     if (cacheSize == 0)
     {
         //
@@ -857,7 +857,7 @@ static size_t GetLogicalProcessorCacheSizeFromOS()
     }
 #endif
 
-#if defined(TARGET_LINUX) && !defined(HOST_ARM)
+#if defined(HOST_ARM64)
     if (cacheSize == 0)
     {
         // It is currently expected to be missing cache size info

--- a/src/coreclr/src/pal/src/misc/sysinfo.cpp
+++ b/src/coreclr/src/pal/src/misc/sysinfo.cpp
@@ -565,7 +565,7 @@ PAL_GetLogicalProcessorCacheSizeFromOS()
     cacheSize = std::max(cacheSize, (size_t)sysconf(_SC_LEVEL4_CACHE_SIZE));
 #endif
 
-#if defined(TARGET_LINUX)
+#if defined(TARGET_LINUX) && !defined(HOST_ARM)
     if (cacheSize == 0)
     {
         //

